Lib/test/test_pathlib/test_pathlib.py

```diff
@@ -1102,6 +1102,7 @@ def test_is_socket_true(self):
         self.assertIs(self.cls(self.base, 'mysock\udfff').is_socket(), False)
         self.assertIs(self.cls(self.base, 'mysock\x00').is_socket(), False)
 
+    @unittest.skipIf(os.name != 'posix', 'test requires a POSIX-compatible system')
```

Comment:
```suggestion
    @needs_posix
```

Reply:
Thank you for the review! I think the `needs_posix` will skip the test if the current test class is not using a posix flavor path class, and they may run on Windows like the `PurePosixPathTest` test class.

In this scenario, we want to skip the test case if the current platform is not posix, so I think the `needs_posix` is not suitable here.

Reply:
`PathTest.test_is_char_device_true()` runs only against `Path` and its subclasses, not `PurePath`. I do think `@needs_posix` is the right thing to do here, and its consistent with other `PathTest` methods.